### PR TITLE
CEDS-4415 Copy declaration when undergo under amendment. Draft implementation

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: app/controllers/AmendDeclarationController.scala [hunk 1]
```diff
@@ -16,40 +16,38 @@
...
-      case Some((_, mrn, lrn, ducr)) => Future.successful(Ok(""))
-      case _                         => errorHandler.displayErrorPage
+  val submit: Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
+    if (!declarationAmendmentsConfig.isEnabled) Future.successful(Redirect(RootController.displayPage))
+    else request.session.get(ExportsSessionKeys.submissionId) match {
+      case Some(submissionId) =>
+        connector.findOrCreateDraftForAmend(submissionId).map { declarationId =>
+          Redirect(SummaryController.displayPage)
+            .addingToSession(ExportsSessionKeys.declarationId -> declarationId)
+        }
```
Comment (udinobi) [inline app/controllers/AmendDeclarationController.scala:48]: We still have a few "submission" keys in Session at this point. Do we need to remove them or we do not care to bring them forward on the next page?

Comment (glidester) [reply app/controllers/AmendDeclarationController.scala]: If I'm understanding you correctly, keeping these session keys active should be fine as this new draft amend declaration is associated with the current submissionId (and will have the same DUCR, LRN and MRN values as it's parent). 
This however is opposed to when you create a completely new declaration by 'copying' an existing declaration. Then the new declaration will have a different submissionId & MRN (once it is submitted) and certainly will have a different DUCR and LRN. In that scenario these session keys should definitely be cleared. :+1:

File: app/controllers/CopyDeclarationController.scala [hunk 2]
```diff
@@ -55,6 +56,7 @@ class CopyDeclarationController @Inject() (
       case Some(submission) if !isDeclarationRejected(submission) =>
         Redirect(CopyDeclarationController.displayPage)
           .addingToSession(ExportsSessionKeys.declarationId -> submissionId)
+          .removingFromSession(submissionDucr, ExportsSessionKeys.submissionId, submissionLrn, submissionMrn)
```
Comment (glidester) [inline app/controllers/CopyDeclarationController.scala:59]: Thanks, good spot! :+1:

**Review by glidester — APPROVED** [review]
LGTM :+1:

**Pull request merged**